### PR TITLE
DOC: match_channel_orders works on Epochs and Evoked, too

### PR DESCRIPTION
**Pull request by sappelhoff**
minor doc contribution. All these classes: Raaw, Epochs, and Evoked work with the "match_channel_orders" function, as it just used the shared `reorder_channels` method under the hood.

This PR clarifies that.

### CODE REVIEW

**Comment by sappelhoff** [comment]
@larsoner do you think we could get this merged?

This is how it looks in rendered state: https://output.circle-artifacts.com/output/job/eb4af659-c944-4ed7-9743-737bc1e36f16/artifacts/0/html/generated/mne.match_channel_orders.html#mne.match_channel_orders

**Comment by mscheltienne** [comment]
Do we need a deprecation cycle for the argument name change?

**Comment by larsoner** [comment]
I think it's very unlikely that users have passed this as a keyword argument. But at the same time it's pretty easy to tack on a `*, raws=None` at the end of the signature and `if raws is not None:` emit a warning and set `insts=raws` or whatever. @sappelhoff I'll let you decide if you think it's worth it or not, it's a coin toss to me :)

**Comment by sappelhoff** [comment]
Thanks for the note and the suggestions. I personally would go ahead without a `raws=None` parameter and without a deprecation cycle, as I agree that it's very unlikely to cause many breakages.

>  But at the same time it's pretty easy to tack on a *, raws=None at the end of the signature and if raws is not None: emit a warning and set insts=raws or whatever.

With this solution:

1. would we keep documenting `raws` in the numpydoc docstring? If so, do we just informally write "this is deprecated, use 'insts' instead"?
2. would we remove this "deprecation hack" after one or two releases? Or would we keep it?

I think if the answer to 2. is "remove it eventually", I would be more prone to adding this "safeguard" :-)

**Comment by sappelhoff** [comment]
ok, I'll add it in this case!

1. not a lot of work
2. may be safer for some
3. not ugly in the long term

thanks

**Comment by sappelhoff** [comment]
the doc build failure seems unrelated:

> E: Unable to locate package python3.12-venv

**Comment by sappelhoff** [comment]
And I have no idea why this test is failing: https://github.com/mne-tools/mne-python/actions/runs/9906155680/job/27367213244?pr=12699#step:17:6028

the DeprecationWarning DOES seem to be raised, but isn't picked up somehow?

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 1]
```diff
@@ -475,11 +475,19 @@ def test_concatenate_raws_order():
 
     with pytest.raises(ValueError, match="Channel order must match."):
         # still fails, because raws is copied and not changed in place
-        match_channel_orders(raws, copy=True)
+        match_channel_orders(insts=raws, copy=True)
         raw_concat = concatenate_raws(raws)
 
+    # XXX: remove in version 1.9
+    with pytest.raises(DeprecationWarning, match="``raws`` parameter is deprecated"):
```
Comment (larsoner) [inline mne/io/fiff/tests/test_raw_fiff.py:482]: Not pytest.raises, pytest.warns

**Review by larsoner — APPROVED** [review]

**Comment by larsoner** [comment]
Pushed a tiny commit to make the tests happy, marking for merge-when-green, thanks in advance @sappelhoff !

File: mne/io/base.py [hunk 2]
```diff
@@ -3083,25 +3083,44 @@ def concatenate_raws(
...
     ----------
-    raws : list
-        List of :class:`~mne.io.Raw` instances to order.
+    insts : list
+        List of :class:`~mne.io.Raw`, :class:`~mne.Epochs`,
+        or :class:`~mne.Evoked` instances to order.
     %(copy_df)s
+    raws : list
+        This parameter is deprecated and will be removed in mne version 1.9.
+        Please use `insts` instead.
```
Comment (larsoner) [inline mne/io/base.py:3097]:
```suggestion
        Please use ``insts`` instead.
```

File: mne/io/fiff/tests/test_raw_fiff.py (hunk 1 above)
Comment (sappelhoff) [reply mne/io/fiff/tests/test_raw_fiff.py]: ohhhhhh thanks @larsoner

**Comment by sappelhoff** [comment]
Thanks for the fixes @larsoner :-)

**Comment by larsoner** [comment]
Thanks @sappelhoff !

**Pull request merged**